src/etc/inc/services.inc

```diff
@@ -1972,14 +1972,29 @@ function dyndnsCheckIP($int) {
 		if (stristr($gateways_status[$config['interfaces'][$int]['gateway']]['status'], "down")) {
 			return "down";
 		}
-		$hosttocheck = "http://checkip.dyndns.org";
+
+		if (is_array($config['checkipservices']['checkipservice'])) {
+			foreach ($config['checkipservices']['checkipservice'] as $i => $checkipservice) {
+				if (isset($checkipservice['enable'])) {
+					$url = $checkipservice['url'];
+					$username = $checkipservice['username'];
+					$password = $checkipservice['password'];
+					$verifysslpeer = isset($checkipservice['verifysslpear']) ? true : false;
+					break;
+				}
+			}
+		}
+
+		$hosttocheck = $url;
```

Comment:
I think the default needs to be hard-coded (probably in globals.inc), and show as such on the checkip page. We might point that to one of our own servers in the near future. Disabling is something that should be done on the individual config entries (though that's only on the RFC 2136 types currently, it should be added to the others as well), not by deleting the services available.